### PR TITLE
[WFCORE-6249] Eliminating deprecated sun.jdk module usages

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.redhat.com/browse/WFCORE-6249

### CODE REVIEW

**Review by yersan — CHANGES_REQUESTED** [review]

File: testsuite/layers/src/test/java/org/jboss/as/test/layers/LayersTestCase.java [hunk 1]
```diff
@@ -36,14 +36,13 @@ public class LayersTestCase {
...
+        // deprecated and unused
         "ibm.jdk",
-        // Un-used
-        "javax.xml.stream.api",
-        // Un-used
+        "javax.api",
         "javax.sql.api",
+        "javax.xml.stream.api",
+        "org.jboss.as.threads",
+        "sun.jdk",
```
[the remark this answers is not in the document]
Comment (ropalka) [reply testsuite/layers/src/test/java/org/jboss/as/test/layers/LayersTestCase.java:45]: You are of course right @yersan . I noticed that too during the weekend when I was doing further WF sun.jdk cleanup.

File: core-feature-pack/common/src/main/resources/modules/system/layers/base/sun/scripting/main/module.xml [hunk 2]
```diff
@@ -23,7 +23,7 @@
   -->
 <module xmlns="urn:jboss:module:1.6" name="sun.scripting">
     <properties>
-        <property name="jboss.api" value="private"/>
+        <property name="jboss.api" value="deprecated"/>
```
Comment (bstansberry) [inline core-feature-pack/common/src/main/resources/modules/system/layers/base/sun/scripting/main/module.xml:26]: I'm pretty sure we should not do this.

The "deprecated" setting means "'public' but be aware we might remove this some time". If this wasn't public before I don't know why it should become public now.

**Review by bstansberry — CHANGES_REQUESTED** [review]

**Comment by ropalka** [comment]
Hello @yersan this PR is awaiting your approval.

**Comment by yersan** [comment]
Hi @ropalka , yes, I'm keeping an eye on it, we need to go with full confidence at this stage and "
Galleon Full integration - Linux" failed twice with your latest changes, apparently it is unrelated, but better to go with all in gree for now

**Review by yersan — APPROVED** [review]

**Comment by yersan** [comment]
Thanks @ropalka

**Pull request merged**